features/matchers/have_enqueued_mail_matcher.feature

```diff
@@ -38,3 +38,40 @@ Feature: have_enqueued_mail matcher
       """
     When I run `rspec spec/mailers/user_mailer_spec.rb`
     Then the examples should all pass
+
+  Scenario: Checking mailer arguments
+    Given a file named "app/mailers/my_mailer.rb" with:
+      """ruby
+      class MyMailer < ApplicationMailer
+
+        def signup(user = nil)
+          @user = user
+
+          mail to: "to@example.org"
+        end
+      end
+      """
+    Given a file named "spec/mailers/my_mailer_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe MyMailer do
+        it "matches with enqueued mailer" do
+          ActiveJob::Base.queue_adapter = :test
+          # Works with plain args
+          expect {
+            MyMailer.signup('user').deliver_later
+          }.to have_enqueued_mail(MyMailer, :signup).with('user')
+          # Works with named parameters
+          expect {
+            MyMailer.with('foo' => 'bar').signup.deliver_later
```

Comment:
Should it be like that? Rails API docs suggest using a symbol for the key:
```ruby
InvitationsMailer.with(inviter: person_a, invitee: person_b).account_invitation.deliver_later
```

Reply:
I think using a hash may induce confusion.
[Rails API example](https://api.rubyonrails.org/classes/ActionMailer/Parameterized.html):
```ruby
class InvitationsMailer < ApplicationMailer
  def account_invitation(inviter, invitee)
...
InvitationsMailer.account_invitation(person_a, person_b).deliver_later
```
Does it make sense to pass a non-hash argument? Is there a difference for the documentation purpose?
Was there a special case for named parameters that you've fixed in this PR? Is it covered sufficiently in specs?

Reply:
I added these examples only because they was missing in relish doc, there's no special case addressed.